src/kernels/MIOpenPadConstant.cpp

```diff
@@ -0,0 +1,190 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <cstddef>
+#include <cstdio>
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_runtime.h>
+#include <hip/hip_fp16.h>
+#endif
+
+#include "float_types.h"
+#include "tensor_view_5d.hpp"
+
+template <typename TI, typename TO>
+__device__ void padconstantfwd(const TI* __restrict__ x,
+                               TO* __restrict__ y,
+                               const tensor_view_5d_t x_tv,
+                               const tensor_view_5d_t y_tv,
+                               const padding_5d_t padding,
+                               const size_t output_size,
+                               TO value)
+{
+    TO padding_value   = CVT_ACCUM2FLOAT(value);
+    const uint64_t gid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(gid >= output_size)
+        return;
+
+    uint64_t o[5];
+    getNCDHW(o, gid, y_tv.size);
+    bool flag = true;
+
+    for(int i = 0; i < 5; ++i)
+    {
+        o[i] = o[i] - padding.val[2 * i];
+        flag *= o[i] < x_tv.size[i];
+    }
+
+    TO val = flag ? get5DValueAt<TO>(x, x_tv.stride, o[0], o[1], o[2], o[3], o[4]) : padding_value;
+    set5DValueAt<TO>(y, y_tv, gid, val);
+}
+
+template <typename TI, typename TO>
+__device__ void padconstantfwdcontiguous(const TI* __restrict__ x,
+                                         TO* __restrict__ y,
+                                         const tensor_view_5d_t x_tv,
+                                         const tensor_view_5d_t y_tv,
+                                         const padding_5d_t padding,
+                                         const size_t output_size,
+                                         TO value)
+{
+    TO padding_value   = CVT_ACCUM2FLOAT(value);
+    const uint64_t gid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(gid >= output_size)
+        return;
+
+    uint64_t o[5];
+    getNCDHW(o, gid, y_tv.size);
+    bool flag = true;
+
+    for(int i = 0; i < 5; ++i)
+    {
+        o[i] = o[i] - padding.val[2 * i];
+        flag *= o[i] < x_tv.size[i];
+    }
+
+    y[gid] = flag ? get5DValueAt<TO>(x, x_tv.stride, o[0], o[1], o[2], o[3], o[4]) : padding_value;
+}
+
+template <typename TI, typename TO>
+__device__ void padconstantbwd(TI* __restrict__ dx,
+                               const TO* __restrict__ y_grad,
+                               const tensor_view_5d_t dx_tv,
+                               const tensor_view_5d_t y_grad_tv,
+                               const padding_5d_t padding,
+                               const size_t input_size)
+
+{
+    const uint64_t gid = threadIdx.x + blockIdx.x * blockDim.x;
+    if(gid >= input_size)
+        return;
+
+    uint64_t o[5];
+    getNCDHW(o, gid, dx_tv.size);
+    bool flag = true;
+
+    for(int i = 0; i < 5; ++i)
+    {
+        o[i] = o[i] + padding.val[2 * i];
+        flag *= o[i] < y_grad_tv.size[i];
```

Comment:
I think that bwd only needs to add a part that checks whether o[i]>=0 after adding the value from o[i].